interface/globals.php

```diff
@@ -653,10 +653,9 @@ function UrlIfImageExists($filename, $append = true)
 // Override temporary_files_dir
 $GLOBALS['temporary_files_dir'] = rtrim(sys_get_temp_dir(), '/');
 
-// turn off PHP compatibility warnings
-ini_set("session.bug_compat_warn", "off");
+error_reporting(error_reporting() & ~E_USER_DEPRECATED & ~E_USER_WARNING);
```

Comment:
should this line go into rel-600, but not master?
(so we fix the underlying warnings in master rather than remove them in order to avoid critical error from creeping up on us)

Reply:
actually, we could maybe do following so then production versions would always do this while -dev would not:
```
if ( $v_tag != "-dev") {
    error_reporting(error_reporting() & ~E_USER_DEPRECATED & ~E_USER_WARNING);
}
```

Reply:
What critical errors? These are user defined which no one pays attention to anyway. Besides, i'd rather see the hard error so it gets taken care of. Besides, it's the wrong way for libraries to deprecate anyway. Should be these same as php does it.
In the end, I don't care.

Reply:
If the trigger_error() mechanism will break things then agree prob best to always ignore these USER things.

Reply:
Agreed. Think about the time that went into tracking this down.
Merging.